packages/xy-chart/src/series/BarSeries.jsx

```diff
@@ -46,24 +45,30 @@ export default class BarSeries extends React.PureComponent {
       onClick,
       onMouseMove,
       onMouseLeave,
+      horizontal,
     } = this.props;
+    if (!xScale || !yScale) return null;
+    const valueScale = horizontal ? xScale : yScale;
+    const categoryScale = horizontal ? yScale : xScale;
+    const barWidth =
+      categoryScale.barWidth || (categoryScale.bandwidth && categoryScale.bandwidth()) || 0;
+    const valueField = horizontal ? x : y;
+    const categoryField = horizontal ? y : x;
 
-    if (!xScale || !yScale || !barWidth) return null;
-
-    const maxHeight = (yScale.range() || [0])[0];
-    const offset = xScale.offset || 0;
+    const maxBarLength = (valueScale.range() || [0])[0];
+    const offset = categoryScale.offset || 0;
 
     return (
       <Group style={disableMouseEvents ? noEventsStyles : null}>
         {data.map((d, i) => {
-          const barHeight = maxHeight - yScale(y(d));
+          const barLength = horizontal ? valueScale(valueField(d)) : maxBarLength - valueScale(valueField(d));
           const color = d.fill || callOrValue(fill, d, i);
-          const barX = xScale(x(d)) - offset;
+          const barPosition = categoryScale(categoryField(d)) - offset;
 
           return (
             isDefined(d.y) && (
```

Comment:
do you think we should update this to be dependent on horizontal? eg `isDefined(horizontal ? d.y : d.x)`

Reply:
@williaster  Actually I'm a bit confused why we need the check here?